ebos/eclgenericproblem_impl.hh

```diff
@@ -367,19 +367,19 @@ updateNum(const std::string& name, std::vector<T>& numbers, std::size_t num_regi
     if (!eclState_.fieldProps().has_int(name))
         return;
 
-    const auto& numData = eclState_.fieldProps().get_int(name);
-
     unsigned numElems = gridView_.size(/*codim=*/0);
-    numbers.resize(numElems);
-    for (unsigned elemIdx = 0; elemIdx < numElems; ++elemIdx) {
-        if (numData[this->getLookUpData(elemIdx)] > (int)num_regions) {
-            throw std::runtime_error("Values larger than maximum number of regions " + std::to_string(num_regions) + " provided in " + name);
-        } else if (numData[this->getLookUpData(elemIdx)] > 0) {
-            numbers[elemIdx] = static_cast<T>(numData[this->getLookUpData(elemIdx)]) - 1;
-        } else {
+    std::function<void(T)> valueCheck = [num_regions,name](T number) {
+        if ( (number+1) > (int)num_regions) {
+            throw std::runtime_error("Values larger than maximum number of regions "
+                                     + std::to_string(num_regions) + " provided in " + name);
+        }
+        if ( (number+1) <= 0) {
```

Comment:
my impression is that this is hard to read. The check is done with the translated (C++ index instead of FORTRAN/Eclipse index). I think there should be comment on that at least.

Maybe it might be clearer to call the functor with untranslated value?